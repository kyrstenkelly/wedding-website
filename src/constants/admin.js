import EventIcon from '@material-ui/icons/Event';
import MailIcon from '@material-ui/icons/Mail';
import PeopleIcon from '@material-ui/icons/People';

export default {
  MENU_ITEMS: [{
    key: 'guests',
    label: 'Guests',
    icon: PeopleIcon
  }, {
    key: 'events',
    label: 'Events',
    icon: EventIcon
  }, {
    key: 'rsvps',
    label: 'RSVPs',
    icon: MailIcon
  }],
  TABLE_COLUMNS: {
    guests: [{
      label: 'First Name',
      key: 'first_name'
    }, {
      label: 'Last Name',
      key: 'last_name'
    }, {
      label: 'Email',
      key: 'email'
    }, {
      label: 'Address',
      key: 'address'
    }, {
      label: 'RSVP Code',
      key: 'rsvp_code'
    }],
    events: [{
      label: 'Name',
      key: 'name'
    }, {
      label: 'Address',
      key: 'address'
    }],
    rsvps: [{
      label: 'Guest',
      key: 'guest'
    }, {
      label: 'Event',
      key: 'event'
    }, {
      label: 'Attending',
      key: 'attending'
    }, {
      label: 'Head Count',
      key: 'head_count'
    }]
  }
}
