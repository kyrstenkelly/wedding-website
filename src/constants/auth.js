export default {
  AUTH0_CLIENT_ID: 'VsVLMjoDeQKdZ6vT4812w8CJa5WcU3HU',
  AUTH0_DOMAIN: 'jamesandkyrsten.auth0.com',
  AUTH0_CALLBACK_URL: window.location.href,
  AUTH0_API_AUDIENCE: 'rsvps-api'
}
