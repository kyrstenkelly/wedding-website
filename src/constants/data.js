const data = {
  weddingDate: '2020-10-10T06',
  content: [{
    title: 'Details',
    key: 'details',
    content: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla scelerisque cursus tortor, commodo cursus diam placerat vel. Fusce porta nulla sed felis pellentesque, id vestibulum mi mollis. Fusce dignissim lorem nec scelerisque hendrerit. Cras rutrum pharetra dolor vel feugiat. Vivamus eleifend posuere ultrices. Etiam consequat odio a aliquet iaculis. In hac habitasse platea dictumst. Maecenas sed luctus est. Duis bibendum urna a tristique volutpat.
      Aliquam elementum vitae elit nec fringilla. Duis rhoncus odio enim, sed dignissim tellus varius iaculis. Aenean suscipit posuere rhoncus. Duis in nisi a nulla gravida aliquam ac viverra magna. Aenean pharetra tincidunt aliquet. Vestibulum tincidunt a tellus a pellentesque. Nam in bibendum velit.
      Etiam consequat posuere mi vel sodales. Morbi blandit nibh tellus, ac vulputate ipsum posuere nec. Quisque feugiat erat vitae urna imperdiet ultricies. Quisque tempor nisi at quam egestas, non ullamcorper ante faucibus. Aenean pretium mauris sed massa faucibus posuere. Praesent scelerisque lorem odio, id aliquet leo egestas in. Nam sed lectus commodo, sollicitudin ex nec, mollis massa. Maecenas varius ut ex id aliquet. Pellentesque ac sodales purus. Donec vel tortor est. Sed rhoncus ante sodales tellus gravida tincidunt. Nulla pulvinar nisi orci.`
  }, {
    title: 'RSVP',
    key: 'rsvp',
    content: 'Find the RSVP code on your invitation! If you happen to lose it, no worries - just shoot us an text or email. :)'
  }, {
    title: 'About Us',
    key: 'about-us',
    content: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla scelerisque cursus tortor, commodo cursus diam placerat vel. Fusce porta nulla sed felis pellentesque, id vestibulum mi mollis. Fusce dignissim lorem nec scelerisque hendrerit. Cras rutrum pharetra dolor vel feugiat. Vivamus eleifend posuere ultrices. Etiam consequat odio a aliquet iaculis. In hac habitasse platea dictumst. Maecenas sed luctus est. Duis bibendum urna a tristique volutpat.
      Aliquam elementum vitae elit nec fringilla. Duis rhoncus odio enim, sed dignissim tellus varius iaculis. Aenean suscipit posuere rhoncus. Duis in nisi a nulla gravida aliquam ac viverra magna. Aenean pharetra tincidunt aliquet. Vestibulum tincidunt a tellus a pellentesque. Nam in bibendum velit.
      Etiam consequat posuere mi vel sodales. Morbi blandit nibh tellus, ac vulputate ipsum posuere nec. Quisque feugiat erat vitae urna imperdiet ultricies. Quisque tempor nisi at quam egestas, non ullamcorper ante faucibus. Aenean pretium mauris sed massa faucibus posuere. Praesent scelerisque lorem odio, id aliquet leo egestas in. Nam sed lectus commodo, sollicitudin ex nec, mollis massa. Maecenas varius ut ex id aliquet. Pellentesque ac sodales purus. Donec vel tortor est. Sed rhoncus ante sodales tellus gravida tincidunt. Nulla pulvinar nisi orci.`
  }]
};

export default data;
