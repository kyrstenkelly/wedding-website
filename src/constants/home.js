import details from '../images/details.jpg';
import detailsCompressed from '../images/details-compressed.jpg';
// import rsvp from '../images/rsvp.jpg';
// import rsvpCompressed from '../images/rsvp-compressed.jpg';
// import story from '../images/story.jpg';
// import storyCompressed from '../images/story-compressed.jpg';
import travel from '../images/travel.jpg';
import travelCompressed from '../images/travel-compressed.jpg';

export default {
  WEDDING_DATE: '2020-10-10T06',
  // TODO: Get events from backend
  EVENTS: [{
    title: 'Ceremony',
    time: '4pm',
    place: 'redwoods',
    address: {
      line1: 'Sanborn County Park',
      line2: '16055 Sanborn Rd',
      line3: 'Saratoga, CA 95070'
    }
  }, {
    title: 'Reception',
    time: '5 - 10pm',
    place: 'redwoods',
    address: {
      line1: 'Saratoga Springs',
      line2: '22801 Big Basin Way',
      line3: 'Saratoga, CA 95070'
    }
  }, {
    time: '7pm - 11pm',
    place: 'austin',
    address: {
      line1: 'Some Cool Venue',
      line2: '123 To Be Determined'
    }
  }],
  HEADER_LINKS: [{
    title: 'Details',
    key: 'details',
    backgroundImage: details,
    backgroundImageCompressed: detailsCompressed
  // }, {
  //   title: 'Story',
  //   key: 'story',
  //   backgroundImage: story,
  //   backgroundImageCompressed: storyCompressed,
  // }, {
  //   title: 'Gallery',
  //   key: 'gallery',
  //   backgroundImage: details,
  //   backgroundImageCompressed: detailsCompressed
  }, {
    title: 'Travel',
    key: 'travel',
    backgroundImage: travel,
    backgroundImageCompressed: travelCompressed
  // }, {
  //   title: 'Gifts',
  //   key: 'gifts',
  //   backgroundImage: travel,
  //   backgroundImageCompressed: travelCompressed
  // }, {
  //   title: 'RSVP',
  //   key: 'rsvp',
  //   backgroundImage: rsvp,
  //   backgroundImageCompressed: rsvpCompressed
  }]
};
