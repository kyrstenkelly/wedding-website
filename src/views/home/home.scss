@import '~shared/styles/common';

.home {
  display: flex;
  flex-direction: column;
  min-height: 100vh;

  .hero {
    position: relative;

    &__overlay {
      background-size: cover;
      position: absolute;
      top: 0;
      left: 0;
      z-index: 1;
      width: 100%;
      height: 100%;

      &:before {
        content: '';
        position: absolute;
        display: block;
        height: 100%;
        width: 100%;
        z-index: 2;
        background-color: rgba(6,24,49, .45);
      }
    }

    &__content {
      position: relative;
      z-index: 3;
      height: 100%;
    }
  }

  .stars-container {
    position: fixed;
    display: flex;
    z-index: 2;
    justify-content: space-between;
    box-sizing: border-box;
    top: 0;
    width: 100vw;
    height: 100%;
    padding: 0 10px;

    .stars-left {
      width: 100%;
      height: 100px;
      margin-top: 70px;
    }

    .stars-right {
      display: none;
    }

    @include media($screen-sm) {
      .stars-left {
        margin-top: 90px;
        height: 175px;
      }
    }

    @include media($screen-md) {
      .stars-left {
        margin-top: 0;
      }

      .stars-right {
        display: initial;
      }

      .stars-left, .stars-right {
        width: calc((100% - #{$max-content-width}) / 2);
        height: 100%;
      }
    }

    // TODO: Add back when we include the header
    // height: calc(100% - 100px);

    // @include media($screen-lg) {
    //   top: 100px;
    // }
  }

  .save-the-date {
    @extend .cursive-font;
    margin-top: rem(70);
    margin-bottom: rem(50);
    font-size: rem(50);
    line-height: rem(60);
    text-align: center;

    p {
      @extend .paragraph-font;
      font-size: rem(20);
      line-height: rem(28);

      &.small {
        font-size: rem(14);
      }
    }
  }

  .content {
    flex: 1;
    background: $darker-blue;
    color: $white;
  }
}
