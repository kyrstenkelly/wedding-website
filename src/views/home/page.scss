@import '~shared/styles/common';

.page {
  display: flex;
  flex-direction: column;
  min-height: 100vh;

  .hero {
    min-height: 400px;

    &__content {
      flex: 1;
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-content: center;
    }

    @include media($screen-md) {
      min-height: 500px;
    }

    @include media($screen-lg) {
      min-height: 600px;
    }

    @include media($screen-xl) {
      min-height: 700px;
    }
  }

  .content {
    flex: 1;
    background: $darker-blue;
    color: $white;
  }
}
