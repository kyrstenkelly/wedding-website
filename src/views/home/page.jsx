import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

import BackgroundImage from 'shared/components/background-image/background-image';
import Countdown from './components/countdown/countdown';
import Footer from 'shared/components/footer/footer';
import Header from './components/header/header';
import constants from '../../constants/home';
import './page.scss';

const { PAGES } = constants;

export class Page extends Component {
  static propTypes = {
    section: PropTypes.oneOf(PAGES.map(l => l.key))
  }

  state = {
    currentSection: PAGES.find(p => p.key === this.props.section),
    backgroundLoaded: false
  }

  componentDidUpdate(prevProps) {
    if (prevProps.section !== this.props.section) {
      this.setState({
        currentSection: PAGES.find(p => p.key === this.props.section)
      });
    }
  }

  render() {
    const { currentSection } = this.state;

    return (
      <div className='page'>
        <BackgroundImage 
          className='hero'
          url={currentSection.backgroundImage}
          placeholderUrl={currentSection.backgroundImageCompressed}
        >
          <Header links={PAGES} />

          <div className='hero__content'>
            <Countdown date={constants.WEDDING_DATE} />
          </div>
        </BackgroundImage>

        { currentSection.component &&
          <div className='content'>
            <div className='contain'>
              {currentSection.component}
            </div>
          </div>
        }

        <Footer theme='dark'></Footer>
      </div>
    );
  }
}

export default withRouter(Page);