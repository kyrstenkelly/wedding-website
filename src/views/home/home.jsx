import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import BackgroundImage from 'shared/components/background-image/background-image';
import Countdown from './components/countdown/countdown';
import Header from './components/header/header';
// import Stars from './components/stars/stars';
import constants from '../../constants/home';
import background from 'images/rsvp.jpg';
import backgroundCompressed from 'images/rsvp-compressed.jpg';
import './home.scss';

export class Home extends Component {
  render() {
    return (
      <div className='home'>
        <BackgroundImage 
          className='hero'
          url={background}
          placeholderUrl={backgroundCompressed}
        >
          {/* <div className='stars-container'>
            <div className='stars-left'>
              <Stars />
            </div>

            <div className='stars-right'>
              <Stars />
            </div>
          </div> */}

          <Header links={constants.PAGES} />

          <div className='hero__content'>
            <Countdown date={constants.WEDDING_DATE} />

            <div className='date'>
              {constants.WEDDING_DATE_FORMATTED}
            </div>
          </div>
        </BackgroundImage>
      </div>
    );
  }
}

export default withRouter(Home);
