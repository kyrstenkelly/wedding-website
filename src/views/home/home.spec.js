import React from 'react';
import { shallow } from 'enzyme';

import { Home } from './home';
import Header from './components/header/header';
import constants from 'constants/home';

// jest.mock('./components/header/header');

describe('<Home />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Home />);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // TODO: Add back when we add the header to the site
  it('generates a list of links from the data', () => {
    const header = wrapper.find(Header);
    expect(header.prop('links')).toEqual(constants.PAGES);
  });
});
