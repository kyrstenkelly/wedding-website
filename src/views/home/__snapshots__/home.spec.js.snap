// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <BackgroundImage
        className="hero"
        placeholderUrl="rsvp-compressed.jpg"
        url="rsvp.jpg"
>
        <Header
                links={
                        Array [
                                Object {
                                  "backgroundImage": "details.jpg",
                                  "backgroundImageCompressed": "details-compressed.jpg",
                                  "component": <Details />,
                                  "key": "details",
                                  "title": "Details",
                                },
                                Object {
                                  "backgroundImage": "travel.jpg",
                                  "backgroundImageCompressed": "travel-compressed.jpg",
                                  "component": <Travel />,
                                  "key": "travel",
                                  "title": "Travel",
                                },
                              ]
                }
        />
        <div
                className="hero__content"
        >
                <Countdown
                        date={"2020-10-10T11:00:00.000Z"}
                />
                <div
                        className="date"
                >
                        Oct 10, 2020
                </div>
        </div>
</BackgroundImage>,
      "className": "home",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header
            links={
                        Array [
                                    Object {
                                      "backgroundImage": "details.jpg",
                                      "backgroundImageCompressed": "details-compressed.jpg",
                                      "component": <Details />,
                                      "key": "details",
                                      "title": "Details",
                                    },
                                    Object {
                                      "backgroundImage": "travel.jpg",
                                      "backgroundImageCompressed": "travel-compressed.jpg",
                                      "component": <Travel />,
                                      "key": "travel",
                                      "title": "Travel",
                                    },
                                  ]
            }
/>,
          <div
            className="hero__content"
>
            <Countdown
                        date={"2020-10-10T11:00:00.000Z"}
            />
            <div
                        className="date"
            >
                        Oct 10, 2020
            </div>
</div>,
        ],
        "className": "hero",
        "placeholderUrl": "rsvp-compressed.jpg",
        "url": "rsvp.jpg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "links": Array [
              Object {
                "backgroundImage": "details.jpg",
                "backgroundImageCompressed": "details-compressed.jpg",
                "component": <Details />,
                "key": "details",
                "title": "Details",
              },
              Object {
                "backgroundImage": "travel.jpg",
                "backgroundImageCompressed": "travel-compressed.jpg",
                "component": <Travel />,
                "key": "travel",
                "title": "Travel",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Countdown
                date={"2020-10-10T11:00:00.000Z"}
/>,
              <div
                className="date"
>
                Oct 10, 2020
</div>,
            ],
            "className": "hero__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "date": "2020-10-10T11:00:00.000Z",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Oct 10, 2020",
                "className": "date",
              },
              "ref": null,
              "rendered": "Oct 10, 2020",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BackgroundImage
          className="hero"
          placeholderUrl="rsvp-compressed.jpg"
          url="rsvp.jpg"
>
          <Header
                    links={
                              Array [
                                        Object {
                                          "backgroundImage": "details.jpg",
                                          "backgroundImageCompressed": "details-compressed.jpg",
                                          "component": <Details />,
                                          "key": "details",
                                          "title": "Details",
                                        },
                                        Object {
                                          "backgroundImage": "travel.jpg",
                                          "backgroundImageCompressed": "travel-compressed.jpg",
                                          "component": <Travel />,
                                          "key": "travel",
                                          "title": "Travel",
                                        },
                                      ]
                    }
          />
          <div
                    className="hero__content"
          >
                    <Countdown
                              date={"2020-10-10T11:00:00.000Z"}
                    />
                    <div
                              className="date"
                    >
                              Oct 10, 2020
                    </div>
          </div>
</BackgroundImage>,
        "className": "home",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Header
              links={
                            Array [
                                          Object {
                                            "backgroundImage": "details.jpg",
                                            "backgroundImageCompressed": "details-compressed.jpg",
                                            "component": <Details />,
                                            "key": "details",
                                            "title": "Details",
                                          },
                                          Object {
                                            "backgroundImage": "travel.jpg",
                                            "backgroundImageCompressed": "travel-compressed.jpg",
                                            "component": <Travel />,
                                            "key": "travel",
                                            "title": "Travel",
                                          },
                                        ]
              }
/>,
            <div
              className="hero__content"
>
              <Countdown
                            date={"2020-10-10T11:00:00.000Z"}
              />
              <div
                            className="date"
              >
                            Oct 10, 2020
              </div>
</div>,
          ],
          "className": "hero",
          "placeholderUrl": "rsvp-compressed.jpg",
          "url": "rsvp.jpg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "links": Array [
                Object {
                  "backgroundImage": "details.jpg",
                  "backgroundImageCompressed": "details-compressed.jpg",
                  "component": <Details />,
                  "key": "details",
                  "title": "Details",
                },
                Object {
                  "backgroundImage": "travel.jpg",
                  "backgroundImageCompressed": "travel-compressed.jpg",
                  "component": <Travel />,
                  "key": "travel",
                  "title": "Travel",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Countdown
                  date={"2020-10-10T11:00:00.000Z"}
/>,
                <div
                  className="date"
>
                  Oct 10, 2020
</div>,
              ],
              "className": "hero__content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "date": "2020-10-10T11:00:00.000Z",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Oct 10, 2020",
                  "className": "date",
                },
                "ref": null,
                "rendered": "Oct 10, 2020",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
